### PR TITLE
Mac: Refactor testShouldHandleFileOpEvent test:

### LINKED ISSUE #1187: Mac: `testShouldHandleFileOpEvent` occasionally fails
We've see the failure below twice now, at a minimum we should add some extra logging/diagnostics to better understand what's happening next time this hits.

```
KauthHandlerTests
  testShouldHandleFileOpEvent, ((testVnodeFile.get() == self->cacheWrapper[ComputeVnodeHashIndex(testVnodeFile.get())].vnode) is true) failed
  /Users/test/devel/VFSForGit/ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm:471

  testShouldHandleFileOpEvent, ((testVnodeFile.get() == self->cacheWrapper[ComputeVnodeHashIndex(testVnodeFile.get())].vnode) is true) failed
  /Users/test/devel/VFSForGit/ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm:487

  testShouldHandleFileOpEvent, ((RootHandle_Indeterminate == self->cacheWrapper[ComputeVnodeHashIndex(testVnodeFile.get())].virtualizationRoot) is true) failed
  /Users/test/devel/VFSForGit/ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm:488
 
  testShouldHandleFileOpEvent, ((testVnodeFile.get() == self->cacheWrapper[ComputeVnodeHashIndex(testVnodeFile.get())].vnode) is true) failed
  /Users/test/devel/VFSForGit/ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm:505
 
  testShouldHandleFileOpEvent, ((rootHandle == self->cacheWrapper[ComputeVnodeHashIndex(testVnodeFile.get())].virtualizationRoot) is true) failed
  /Users/test/devel/VFSForGit/ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm:506 
 
       Executed 92 tests, with 5 failures (0 unexpected) in 0.196 (0.229) seconds
```

### PR DESCRIPTION
**Pull request by wilbaker**
Resolves #1187

  - Split testShouldHandleFileOpEvent into separate test cases with more descriptive names
  - Fix an issue where the `testShouldHandleFileOpEvent_VnodeCacheUpdated` tests were sometimes failing due to vnode cache collisions from other tests
  - Updated the provider initiated I/O and provider offline scenarios to use file vnodes rather than the repo root vnodes

### CODE REVIEW

File: ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm [hunk 1]
```diff
@@ -396,19 +389,26 @@ - (void)testShouldHandleFileOpEvent {
             true, // isDirectory,
             &testRootHandle,
             &pid));
+}
 
-    // Invalid File System should fail
-    XCTAssertFalse(
```
Comment (wilbaker) [inline ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: This case was inadvertently dropped

File: ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm [hunk 2]
```diff
@@ -376,6 +368,7 @@ - (void)testShouldHandleFileOpEvent {
             &testRootHandle,
             &pid));
 
+    org_vfsforgit_PrjFSProviderUserClient userClient;
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: I realise the old code did this too, but the `#define`d name, `PrjFSProviderUserClient` is generally preferable in code - that's the whole point of the macro, if the reverse DNS prefix changes, you don't have to update all of the code.

Might be worth doing a search for the use of the explicitly-prefixed name, I notice quite a few instances of it being used in this file.

File: ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm [hunk 3]
```diff
@@ -420,40 +440,94 @@ - (void)testShouldHandleFileOpEvent {
...
-    MockProcess_AddProcess(501 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+}
+
+- (void)testShouldHandleFileOpEvent_VnodeCacheUpdated {
+    PerfTracer perfTracer;
+    shared_ptr<mount> testMount = mount::Create();
+    std::string repoPath = "/Users/test/code/Repo";
+    shared_ptr<vnode> repoRootVnode = testMount->CreateVnodeTree(repoPath, VDIR);
+    shared_ptr<vnode> testVnodeFile = testMount->CreateVnodeTree(repoPath + "/file.txt");
+    shared_ptr<vnode> testVnodeDirectory = testMount->CreateVnodeTree(repoPath + "/directory", VDIR);
```
Comment (pmj) [inline ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: A lot of this code seems to be repeated in each of the newly-split tests. I guess that's why it was one giant test before.

I'm wondering if it'd be better to place these similar tests into an entirely new class where the boilerplate objects can be placed in ivars, and initialised/shut down in `- (void)setUp`/`- (void)tearDown`.

**Comment by wilbaker** [comment]
@pmj thanks for the feedback, I've filed #1195 to make additional changes based on your feedback and will have a follow up PR with the changes.

**Pull request merged**